lib/publish.js

```diff
@@ -1,43 +1,66 @@
 const execa = require('execa')
+const semver = require('semver')
 
 /**
  * @typedef {import('./index').Context} Context
  */
 /**
- * Push new docker image.
+ * Tag and push new docker images with semantic versioning.
  *
  * @param {Object} pluginConfig - The plugin configuration.
  * @param {Context} context - The semantic-release context.
  * @returns {Promise} A `Promise` that resolve to docker push command.
  */
 module.exports = async (pluginConfig, { nextRelease: { version }, logger }) => {
-  try {
+  /**
+   * Tag and push docker image by version.
+   *
+   * @param {string} version - Tag version.
+   * @returns {Promise} A `Promise` that resolve to docker push command.
+   */
+  async function publish (version) {
     logger.log(
       `Pushing version ${
         process.env.CI_REGISTRY_IMAGE
       }:${version} to GitLab Container Registry`
     )
-
-    // Push both new version and latest
+    // tag image
     await execa(
       'docker',
-      [
-        'tag',
-        `${process.env.CI_REGISTRY_IMAGE}:latest`,
-        `${process.env.CI_REGISTRY_IMAGE}:${version}`
-      ],
+      ['tag', `${process.env.CI_REGISTRY_IMAGE}:${version}`],
```

Comment:
It's not necessary add reference to source image (`latest`) in [tag command](https://docs.docker.com/engine/reference/commandline/tag/)?

Reply:
Could you elaborate?
I removed the latest tag from, as if it's not a prod-release, such as alpha, it should not be tagged with the latest tag.

Reply:
You must remember that the only thing that the docker tag command does is create an alias for a base image. And when the `build` command is executed without indicating any tag (`docker build -t $CI_REGISTRY_IMAGE .`), the `latest` is used by default. Therefore the default base image is latest and it is this that should be used as a source for the docker tag command used in the library. A example in bash.

```sh
CI_REGISTRY_IMAGE=registry.gitlab.com/my-group/my-app
docker build -t $CI_REGISTRY_IMAGE .
# Tag and push v1
docker tag $CI_REGISTRY_IMAGE $CI_REGISTRY_IMAGE:1
docker push $CI_REGISTRY_IMAGE:1
# Tag and push v1.0
docker tag $CI_REGISTRY_IMAGE $CI_REGISTRY_IMAGE:1.0
docker push $CI_REGISTRY_IMAGE:1.0
# Tag and push v1.0.0
docker tag $CI_REGISTRY_IMAGE $CI_REGISTRY_IMAGE:1.0.0
docker push $CI_REGISTRY_IMAGE:1.0.0
# Tag and push v1.0.0-beta
docker tag $CI_REGISTRY_IMAGE $CI_REGISTRY_IMAGE:1.0.0-beta
docker push $CI_REGISTRY_IMAGE:1.0.0-beta
# Tag and push beta
docker tag $CI_REGISTRY_IMAGE $CI_REGISTRY_IMAGE:beta
docker push $CI_REGISTRY_IMAGE:beta
```

Reply:
Think I got what you mean!

Many thanks for the detailed answer, learned something!
So as I understand, the default docker build creates a tag with :latest.
Now, we specify the source build as :latest and tag it with something new.
In case, it is pushing to the latest tag, we do not have to tag it again with :latest, that's why the if